openweather/src/main/java/action/PlaceOrderAction.java

```diff
@@ -0,0 +1,96 @@
+package action;
+
+import org.openqa.selenium.WebElement;
+import pages.MarketPlacePage;
+import pages.PlaceOrderPage;
+
+import java.util.List;
+
+public class PlaceOrderAction extends ActionBase {
+    private final MarketPlacePage marketPlacePage;
+    private final PlaceOrderPage placeOrderPage;
+
+    public PlaceOrderAction() {
+
+        super();
+        marketPlacePage = getPageInstance(MarketPlacePage.class);
+        placeOrderPage = getPageInstance(PlaceOrderPage.class);
+    }
+
+    public PlaceOrderAction openMarketPlace() {
+        marketPlacePage.open();
+
+        return this;
+    }
+
+    public PlaceOrderAction selectProduct(String product) {
+        marketPlacePage.selectProductToOrder(product);
+
+        return this;
+    }
+
+    public PlaceOrderAction addLocation(String searchStr, String searchBy) {
```

Comment:
Using string parameters is the most common problem of QAs, **DON'T DO THIS**.  It definitely leads to bad design. As I mentioned in the above comment, you define Product obj and it's done. Product obj should have all properties such as State, year, and so on.

Reply:
Got it